server/src/main/java/org/elasticsearch/plugins/ActionPlugin.java

```diff
@@ -87,6 +88,24 @@ default Collection<RestHandler> getRestHandlers(
         return Collections.emptyList();
     }
 
+    /**
+    * Rest handlers added by this plugin with ThreadContext.
+    */
+    default Collection<RestHandler> getRestHandlers(
```

Comment:
I get that this is needed to get the ThreadContext as a constructor argument, but it looks odd that we are adding a second getRestHandlers method, and I don't follow where this is called e.g. in ActionModule. I must be missing something.

Reply:
I removed ThreadContext from the constructors and added a method to set it in the parent class. I think this is cleaner and avoids the duplication of getRestHandler.